src/kmergraph.cpp

```diff
@@ -238,20 +218,271 @@ void KmerGraph::discover_k() {
+    // false if have different numbers of nodes
+    if (y.nodes.size() != nodes.size()) {//cout << "different numbers of nodes" << endl;
+        return false;
+    }
+
+    // false if have different nodes
+    for (const auto &kmer_node_ptr: nodes) {
+        const auto &kmer_node = *kmer_node_ptr;
+        // if node not equal to a node in y, then false
+        auto found = find_if(y.nodes.begin(), y.nodes.end(), condition(kmer_node.path));
+        if (found == y.nodes.end()) {
+            return false;
+        }
+
+        // if the node is found but has different edges, then false
+        if (kmer_node.outNodes.size() != (*found)->outNodes.size()) { return false; }
+        if (kmer_node.inNodes.size() != (*found)->inNodes.size()) { return false; }
+        for (uint32_t j = 0; j != kmer_node.outNodes.size(); ++j) {
+            if ((*found)->findNodeInOutNodes(*(kmer_node.outNodes[j].lock())) == (*found)->outNodes.end()) {
+                return false;
+            }
+        }
+
+    }
+    // otherwise is true
+    return true;
+}
+
+bool pCompKmerNode::operator()(KmerNodePtr lhs, KmerNodePtr rhs) {
+    return (lhs->path) < (rhs->path);
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//KmerGraph methods definitions
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//KmerGraphWithCoverage methods definitions
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+void KmerGraphWithCoverage::set_exp_depth_covg(const uint32_t edp) {
     assert(edp > 0);
     exp_depth_covg = edp;
 }
 
-void KmerGraph::set_p(const float e_rate) {
+void KmerGraphWithCoverage::set_p(const float e_rate) {
```

Comment:
Could `p` be named something differently? Again, I have no sense of what it is from reading the code.

Reply:
`p` is the binomial probability parameter

Reply:
updated in commit bec1ff33d23fb4178752ee6a586f07ccd6d9a0d0